hedera-mirror-importer/src/test/java/com/hedera/mirror/importer/addressbook/AddressBookServiceImplTest.java

```diff
@@ -971,30 +972,83 @@ void getNodesWithNodeStake() {
+
+        assertThat(customAddressBookService.getNodes())
+                .hasSize(TEST_INITIAL_ADDRESS_BOOK_NODE_COUNT)
+                .allSatisfy(c -> assertThat(c).returns(4L, ConsensusNode::getTotalStake))
+                .allSatisfy(c -> assertThat(c.getPublicKey()).isNotNull())
+                .allMatch(c -> c.getStake() == 1L)
+                .extracting(ConsensusNode::getNodeId)
+                .containsExactly(0L, 1L, 2L, 3L);
     }
 
     @Test
     void getNodesWithNodeStakeCountMoreThanAddressBook() {
+        MirrorProperties otherMirrorProperties = new MirrorProperties();
+        otherMirrorProperties.setConsensusMode(ConsensusMode.STAKE);
+        AddressBookService customAddressBookService = new AddressBookServiceImpl(addressBookRepository,
+                fileDataRepository, otherMirrorProperties, nodeStakeRepository, transactionTemplate);
         long timestamp = domainBuilder.timestamp();
         domainBuilder.nodeStake().customize(n -> n.consensusTimestamp(timestamp).nodeId(0).stake(0L)).persist();
         domainBuilder.nodeStake().customize(n -> n.consensusTimestamp(timestamp).nodeId(1).stake(0L)).persist();
         domainBuilder.nodeStake().customize(n -> n.consensusTimestamp(timestamp).nodeId(2).stake(0L)).persist();
         domainBuilder.nodeStake().customize(n -> n.consensusTimestamp(timestamp).nodeId(3).stake(0L)).persist();
         domainBuilder.nodeStake().customize(n -> n.consensusTimestamp(timestamp).nodeId(4).stake(0L)).persist();
 
+        assertThat(customAddressBookService.getNodes())
+                .hasSize(TEST_INITIAL_ADDRESS_BOOK_NODE_COUNT)
+                .allSatisfy(c -> assertThat(c).returns(5L, ConsensusNode::getTotalStake))
+                .allSatisfy(c -> assertThat(c.getPublicKey()).isNotNull())
+                .allMatch(c -> c.getStake() == 1L)
+                .extracting(ConsensusNode::getNodeId)
+                .containsExactly(0L, 1L, 2L, 3L);
+
+        // consensusMode == ConsensusMode.STAKE_IN_ADDRESS_BOOK (the default)
+        // here we only include nodes that are in the address book (which is of size only 4, hence total stake = 4)
         assertThat(addressBookService.getNodes())
                 .hasSize(TEST_INITIAL_ADDRESS_BOOK_NODE_COUNT)
+                .allSatisfy(c -> assertThat(c).returns(4L, ConsensusNode::getTotalStake))
+                .allSatisfy(c -> assertThat(c.getPublicKey()).isNotNull())
                 .allMatch(c -> c.getStake() == 1L)
-                .allMatch(c -> c.getTotalStake() == 5L)
+                .allMatch(c -> c.getTotalStake() == 4L)
                 .allMatch(c -> c.getNodeAccountId().getEntityNum() - 3 == c.getNodeId())
-                .allSatisfy(c -> assertThat(c.getPublicKey()).isNotNull());
+                .extracting(ConsensusNode::getNodeId)
+                .containsExactly(0L, 1L, 2L, 3L);
+    }
+
+    @Test
+    void getNodesWithNodesNotInAddressBook() {
```

Comment:
We should be able to consolidate the logic to one parameterized test.
```suggestion
    @CsvSource(textBlock = """
    EQUAL, 6, 6
    ...
    """)
    @ParameterizedTest
    void getNodes(ConsensusMode mode, long stake, int nodeCount) {
```

Reply:
Done.